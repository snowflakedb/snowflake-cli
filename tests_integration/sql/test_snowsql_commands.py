import pytest


@pytest.fixture
def existing_query_id(runner):
    result = runner.invoke_with_connection_json(["sql", "-q", "select 15;>"])
    assert result.exit_code == 0, result.output
    yield result.json[0]["scheduled query ID"]


@pytest.mark.integration
def test_queries(runner):
    query = """select 1;
    select 3;
    select 13;
    !queries;
    select 4;"""
    result = runner.invoke_with_connection(["sql", "-q", query])
    assert result.exit_code == 0, result.output
    assert result.output.count("select 13") == 2
    assert result.output.count("select 3") == 2
    assert result.output.count("select 4") == 1
    assert result.output.count("SUCCEEDED") == 3
    for header in ["QUERY ID", "SQL TEXT", "STATUS", "DURATION_MS"]:
        assert header in result.output


@pytest.mark.integration
def test_queries_help(runner, snapshot):
    result = runner.invoke_with_connection(["sql", "-q", "!queries help;"])
    assert result.exit_code == 0, result.output
    assert result.output == snapshot


@pytest.mark.integration
def test_result(runner, existing_query_id):
    result = runner.invoke_with_connection_json(
        ["sql", "-q", f"!result {existing_query_id}"]
    )
    assert result.exit_code == 0, result.output
    assert result.json == [{"15": 15}]


@pytest.mark.integration
def test_abort(runner, existing_query_id):
    result = runner.invoke_with_connection_json(
        ["sql", "-q", f"!abort {existing_query_id}"]
    )
    assert result.exit_code == 0, result.output
    assert result.json == [
        {
            f"SYSTEM$CANCEL_QUERY('{existing_query_id.upper()}')": "Identified SQL statement is not currently executing.",
        },
    ]
