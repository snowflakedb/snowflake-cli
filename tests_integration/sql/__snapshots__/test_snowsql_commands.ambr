# serializer version: 1
# name: test_queries_help
  '''
  +--------------------------------------------+
  | FILTER    | ARGUMENT             | DEFAULT |
  |-----------+----------------------+---------|
  | amount    | integer              | 25      |
  | status    | string               | any     |
  | warehouse | string               | any     |
  | user      | string               | any     |
  | start     | date in milliseconds | any     |
  | end       | date in milliseconds | any     |
  | type      | string               | any     |
  | duration  | time in milliseconds | any     |
  | session   | No arguments         | any     |
  +--------------------------------------------+
  
  '''
# ---
